### ISSUE TITLE
convert_to_async skips custom awaitables (GetAwaiter present, name ≠ Task/ValueTask) → NoAsyncCalls/3081

### ISSUE THREAD

**Issue by JoshuaRamirez** [JoshuaRamirez/RoslynMcpServer#1170]
## Summary

`convert_to_async` decides eligibility with `IsAwaitableType`, which is **string-prefix only** (`Task` / `ValueTask` display names). Types that are legal C# awaitables via duck-typed instance `GetAwaiter()` (awaiter with `IsCompleted` / `GetResult` / `INotifyCompletion`) but are **not** named Task/ValueTask are never selected by `FindAwaitableCalls`. Eligibility then throws **`ErrorCodes.NoAsyncCalls` (3081)** — “Method has no awaitable calls to convert.” — even when `await Inner();` would compile.

This is the complementary **false-negative** side of string-prefix awaitable detection. Sibling `#1163` is the **false-positive** side (non-awaitable `Task*` types such as `TaskCompletionSource` / `TaskFactory` get awaited → CS1061). A general GetAwaiter / `IsAwaitable` check would close both classes.

Eligibility fails closed (`awaitableCalls.Count == 0` → 3081), so the tool refuses conversion instead of reporting SUCCESS with broken code.

## Repro

Master `898b07ed750541961809bc6ad5f781c0ee2440e9` (Release **v0.6.2**, packages Roslyn **5.0.0**).

```csharp
using System;
using System.Runtime.CompilerServices;

struct MyAwaiter : INotifyCompletion
{
    public bool IsCompleted => true;
    public void OnCompleted(Action continuation) { }
    public int GetResult() => 42;
}
struct MyAwaitable { public MyAwaiter GetAwaiter() => new MyAwaiter(); }

class Worker
{
    public MyAwaitable Inner() => new MyAwaitable();
    public void UseCustom() { Inner(); }  // convertible: await Inner();
    public void UseTask() { System.Threading.Tasks.Task.Delay(1); } // control OK
}
```

1. `convert_to_async` on `UseCustom` → **ErrorCode=3081** / “Method has no awaitable calls to convert.” (**WRONG** — false negative).
2. Ideal body `async Task UseCustomAsync() { await Inner(); }` compiles clean.
3. Control: `convert_to_async` on `UseTask` → Preview **SUCCESS**.

Live probe (Roslyn **5.0.0**): product `IsAwaitableType` copy → `Inner()` return `MyAwaitable`, `productIsAwaitable=False`, ideal GetAwaiter=`True` → product eligibility **NoAsyncCalls**; ideal would convert (1). Real product op (`ConvertToAsyncOperation` + MSBuild workspace): `UseCustom` → **3081 NoAsyncCalls WRONG=yes**; `UseTask` Preview Success=True **CONTROL_OK=yes**.

## Expected

- Types with instance `GetAwaiter()` (duck-typed awaitable / `INotifyCompletion` awaiter) are eligible for `convert_to_async` when the call is a legal await target (e.g. expression-statement `Inner();` → `await Inner();`).
- Do not refuse with **NoAsyncCalls/3081** solely because the type name is not `Task`/`ValueTask`.
- Prefer symbol/GetAwaiter-based awaitable detection (also closes complementary `#1163` Task* prefix false positives if done generally).
- Real `Task` / `Task<T>` / `ValueTask` / `ValueTask<T>` still convert as today.

## Actual

| Shape | Product | Result | Expected |
| --- | --- | --- | --- |
| `void UseCustom() { Inner(); }` (`MyAwaitable`) | **3081 NoAsyncCalls** | refuse | convert / `await Inner();` |
| Ideal `async Task UseCustomAsync() { await Inner(); }` | (manual) | compiles | unchanged intent |
| `void UseTask() { Task.Delay(1); }` | Preview SUCCESS | OK | unchanged intent |

## Root cause

`IsAwaitableType` (~L1160–1168) is **string-prefix only**:

```csharp
return typeName.StartsWith("System.Threading.Tasks.Task") ||
       typeName.StartsWith("System.Threading.Tasks.ValueTask") ||
       typeName.StartsWith("Task<") ||
       typeName.StartsWith("ValueTask<") ||
       typeName == "Task" ||
       typeName == "ValueTask";
```

**Zero GetAwaiter / INotifyCompletion awareness** on this path. `FindAwaitableCalls` (~L1137–1157) uses that check; eligibility (~L170–175) throws `ErrorCodes.NoAsyncCalls` (**3081**) when count is 0. Legal custom awaitables are never selected → conversion refused while `await Inner();` would compile.

## Acceptance criteria

- [ ] Types with instance `GetAwaiter()` (duck-typed awaitable / `INotifyCompletion` awaiter) are eligible for `convert_to_async` when the call is a legal await target (e.g. expression-statement `Inner();` → `await Inner();`)
- [ ] Do not refuse with **NoAsyncCalls/3081** solely because the type name is not `Task`/`ValueTask`
- [ ] Prefer symbol/GetAwaiter-based awaitable detection (also closes complementary `#1163` Task* prefix false positives if done generally)
- [ ] Real `Task` / `Task<T>` / `ValueTask` / `ValueTask<T>` still convert as today
- [ ] allFiles uses the same rule
- [ ] Regression: custom awaitable expression-statement converts; Task control still converts; non-awaitable Task* still skipped (see `#1163`)
- [ ] Does not claim to fix `#1163` FP prefix CS1061, `#1169`/`#1166`/`#1160` typed/var await CS*, `#1156` Wait/Result/GetResult, `#1153` Task\<Task\> double-wrap unless deliberately batched

## Out of scope

- Non-awaitable `Task*` prefix false positives (TCS/TaskFactory/…) → CS1061 — `#1163`
- Typed / field / arg await of generic Task\<T\> / ValueTask\<T\> → CS0029/CS1503 — `#1169`
- Non-generic typed / field / arg await → void CS0029/CS1503 — `#1166`
- `var` / discard await of non-generic Task/ValueTask → CS0815/CS8209 — `#1160`
- Sync-block `.Wait` / `.Result` / `GetAwaiter().GetResult()` — `#1156`
- Already-Task return double-wrap / `return await` — `#1153`
- Leftover extract work `#1167` — label **bug** only

## Separate follow-ups (noticed, NOT filed)

1. **`.ConfigureAwait(false).GetAwaiter().GetResult()`** — still deferred; **#1156 OPEN** (`#1169` FU2 / `#1166` FU3 / `#1160` FU1 / `#1156` FU2).
2. Once eligibility is fixed: blindly awaiting a custom awaitable into a **typed `MyAwaitable` local** may need parent-context rules (same class as `#1169`/`#1166`) — speculative until GetAwaiter eligibility lands.
3. DocumentNotEditable / reserved / allFiles+line for this tool remain `#552` / `#571` / `#489`.

## Named from

`#1169` Separate follow-ups item 1 (also `#1163` FU2 / `#1153` FU2 / `#1166` FU2). Conditions met: live-verified **WRONG=yes** **and** no GetAwaiter-aware fix in the code path (still prefix-only on master `898b07e…`).

## Evidence footer

- Master: `898b07ed750541961809bc6ad5f781c0ee2440e9` (Extract MethodSymbolHelpers `#1164`/`#1165`; Release **v0.6.2**)
- Packages: Roslyn **5.0.0** / product **0.6.2**
- Product path: `IsAwaitableType` ~L1160–1168 (string-prefix only); `FindAwaitableCalls` ~L1137–1157; eligibility ~L170–175 → `ErrorCodes.NoAsyncCalls` (**3081**)
- Live probe: Roslyn **5.0.0** — product `IsAwaitableType` on `MyAwaitable` → False; ideal GetAwaiter → True; real `ConvertToAsyncOperation` on `UseCustom` → **3081 WRONG=yes**; `UseTask` control Preview Success=True **CONTROL_OK=yes**; ideal `async Task UseCustomAsync() { await Inner(); }` compiles clean
- Open leftover at filing: **#1167** — label **bug** only; not leftover (not exclusive-end class)
- Closest (orthogonal): `#1163` = complementary **false-positive** Task* prefix → CS1061 (out-of-scope FU2 = this); `#1156` = mis-await of sync-block on real Tasks; `#1169`/`#1166`/`#1160`/`#1153` = SUCCESS + illegal await shapes on Task/ValueTask — **none** is convert_to_async **NoAsyncCalls/3081** for custom GetAwaiter awaitables
- Leave Dependabot untouched
